runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegisterView/UsersRegisterView.tsx

```diff
@@ -1,9 +1,147 @@
+
+import { DefaultPageValue, ROWS_PER_PAGE } from '../UsersBrowseView/UsersBrowseView.utils';
+import UsersRegisterTable from './UsersRegisterTable';
+import { StyledButtonsContainer, StyledButton, DeleteButton, StyledActionsContainer, StyledTextField } from './UsersRegisterView.styles';
+
+interface SelectedRoles { [id: number]: Role };
+interface MapActivatedUserParams {
+    id: number;
+    login: string;
+    roles: Role[];
+    activated: boolean;
+}
+
+const UsersRegisterView: FC = () => {
+    const { enqueueSnackbar } = useSnackbar();
+    const { translate } = useTranslations();
+    const dispatch = useDispatch();
+    const router = useRouter();
+    const searchParams = useSearchParams();
+
+    const currentPage = searchParams.get('page');
+    const pageSizeFromUrl = searchParams.get('pageSize');
+    const [page, setPage] = useState(currentPage ? parseInt(currentPage, 10) : DefaultPageValue.PAGE);
+    const [limit, setLimit] = useState(
+        pageSizeFromUrl && ROWS_PER_PAGE.includes(parseInt(pageSizeFromUrl, 10))
+            ? parseInt(pageSizeFromUrl, 10)
+            : DefaultPageValue.PAGE_SIZE);
+
+    const { notActivated } = useSelector(usersSelector);
+    const { search, handleSearch, refreshSearch } = useUserSearch(false, limit, page);
+
+    const [selectedRoles, setSelectedRoles] = useState<SelectedRoles>({});
+    const [selections, setSelections] = useState<number[]>([]);
+
+    const handleSelectedRolesChange = (id: number, value: Role) =>
+        setSelectedRoles({ ...selectedRoles, [id]: value});
+
+    const handleSelectionChange = (selection) => setSelections(selection);
+
+    const handleAccept = () => {
+        const hasEveryUserSelectedRole = selections.every(checkUserRole);
+        if (hasEveryUserSelectedRole) {
+            const payload = selections.map((userId) => {
+                const role = selectedRoles[userId];
+                return mapActivatedUserRequest(userId, role);
+            });
+
+            handleSubmit(payload);
+        } else {
+            enqueueSnackbar(translate('Users.Register.View.Events.Error.RolesNotSelected'), { variant: 'error' });
+        }
+    };
+
+    const mapActivatedUserRequest = (id: number, role: Role): MapActivatedUserParams => {
+        const { login } = notActivated.allByPage.content.find((row) => row.id === id);
+        return { id, login, roles: [role], activated: true };
+    };
+
+    const handleSubmit = (params: IUser[]) =>
+        Promise.allSettled(
+            params.map((param) => dispatch(usersActions.updateNotActivated(param)))
+        ).then(() => {
+            enqueueSnackbar(translate('Users.Register.View.Events.Success.AcceptingUser'), { variant: 'success' });
+            refreshSearch();
+        }
+        ).catch((err) => {
+            enqueueSnackbar(translate('Users.Register.View.Events.Error.AcceptFailed'), { variant: 'error' });
+            throw err;
+        });
```

Comment:
imo this should be formatted differently

Promise
    .allSettled(
        params.map((param) => dispatch(usersActions.updateNotActivated(param))))
    .then(() => {
        enqueueSnackbar(translate('Users.Register.View.Events.Success.AcceptingUser'), { variant: 'success' });
        refreshSearch();
    })
    .catch((err) => {
        enqueueSnackbar(translate('Users.Register.View.Events.Error.AcceptFailed'), { variant: 'error' });
        throw err;
    });

<img width="521" alt="image" src="https://github.com/runbotics/runbotics/assets/69820069/89327066-a9a6-41d5-800d-e254c4b0e5bb">

Reply:
also remove throw error, as it would break the whole app as far as i kno